src/lib/pubkey/sphincsplus/sphincsplus_common/sphincsplus.cpp

```diff
@@ -19,14 +20,45 @@
 #include <botan/internal/sp_xmss.h>
 #include <botan/internal/stl_util.h>
 
-#if !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2) and !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE)
+#include <utility>
+
+#if !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2) and !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE) and \
+   !defined(BOTAN_HAS_SLH_DSA_WITH_SHA2) and !defined(BOTAN_HAS_SLH_DSA_WITH_SHAKE)
 static_assert(
    false,
-   "botan module 'sphincsplus_common' is useful only when enabling at least 'sphincsplus_sha2' or 'sphincsplus_shake'");
+   "botan module 'sphincsplus_common' is useful only when enabling at least 'sphincsplus_sha2', 'sphincsplus_shake', 'slh_dsa_sha2', or 'slh_dsa_shake'");
 #endif
 
 namespace Botan {
 
+namespace {
+// FIPS 205, Algorithm 22, line 8
+SphincsMessageInternal prepare_message(SphincsInputMessage msg,
+                                       const Sphincs_Parameters& params,
+                                       StrongSpan<const SphincsContext> context) {
+   BOTAN_ARG_CHECK(params.is_slh_dsa() || context.empty(), "Context is not supported for SPHINCS+");
+#if defined(BOTAN_HAS_SLH_DSA_WITH_SHA2) || defined(BOTAN_HAS_SLH_DSA_WITH_SHAKE)
+   if(params.is_slh_dsa()) {
+      // prefix (no pre-hash mode): input mode byte + |ctx| + ctx
+      const uint8_t input_mode_byte = 0x00;  // Pure (TODO: pre-hash mode: 0x01)
+      return {.prefix = concat<SphincsMessagePrefix>(
+                 store_be(input_mode_byte), store_be(checked_cast_to<uint8_t>(context.size())), context),
+              .message = std::move(msg)};
```

Comment:
If you put the extra comma behind the `std::move` it'll come up with a nicer formatting:

```suggestion
              .message = std::move(msg),};
```

(not in the suggestion above)